### PR TITLE
Fixes #20883: Built in Techniques using generic methods have no reports with new report id

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/20883

### CODE REVIEW

**Comment by amousset** [comment]
This PR breaks webapp tests

**Comment by amousset** [comment]
LGTM appart from that

**Comment by VinceMacBuche** [comment]
Commit modified

**Review by amousset — APPROVED** [review]

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala [hunk 1]
```diff
@@ -183,8 +183,9 @@ object BuildBundleSequence {
     , enableMethodReporting  : Boolean
   ) {
     val contextBundle : List[Bundle]  = main.map(_.id).distinct.collect{ case Some(id) =>
-      Bundle(None, BundleName("rudder_reporting_context"), List((id.directiveId.serialize,"directiveId"), (id.ruleId.serialize, "ruleId"), (techniqueId.name.value,"techniqueName")).map( (BundleParam.DoubleQuote.apply _).tupled ) )
-    }
+      Bundle(None, BundleName("rudder_reporting_context"), List((id.directiveId.serialize,"directiveId"), (id.ruleId.serialize, "ruleId"), (techniqueId.name.value,"techniqueName")).map( (BundleParam.DoubleQuote.apply _).tupled ) ) ::
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala:186]: you could have used rudder_reporting_context_v4 https://github.com/Normation/ncf/pull/1340/files#diff-0bdc2ac14352d653dea5f8de09cff543e70c7ffa33e8b8e82a1388dab7c938c2R536

**Pull request merged**